### ISSUE TITLE
Planning of Home Assistant Integration

### ISSUE THREAD

**Issue by smoki3** [dknowles2/pydrawise#33]
Hi,

I saw you are planning to integrate pydrawise into HA Core. Do you have any plan to release a HACS custom integration for this? Because I think until they approve the new code it will take much time...

**Comment by dknowles2** [comment]
I considered it, but getting it integrated back into core will be difficult if I go that route.

I'm planning to start working on this again this week; I've been distracted with real-life stuff lately. :-)

**Comment by jjhuff** [comment]
Let me know when you're ready for a tester. I have a Pro-HC setup with a flow meter that I'd love to monitor!

**Comment by dknowles2** [comment]
Okay, so here's the current status:

Config flow was finally merged, which allowed me to add devices. In 2023.10, the Hydrawise integration will finally look more like a modern integration.

I've got a PR pending to add more tests to the integration, and once that's merged, I have another fix for entities not loading their state until the first refresh.

I added another implementation of the legacy API that has the same interface as my new GraphQL client, and I have pending changes to update the integration to use that. One that's done, there's one more PR to migrate to the new GraphQL client that uses a different auth method. I got that working this morning, but still need to update tests.

Assuming PRs get merged quickly from now on, I think this could all land before the 2023.11 release, but I've got a bunch of other stuff going on in October, so I wouldn't be surprised if it slips to the 2023.12 release.

Once all this lands, we can start adding new features and fixing other weird stuff. Top of mind for me:

* The controller connectivity binary_sensor isn't correct. It needs to read status from the API, but the signal was removed from the old API, so as a temporary hack we're just looking at whether the last refresh was successful.
* The next cycle sensor needs fixing. If there's an active suspension, the value continually jumps to some point in the future. I think it should just not show a value if the zone is suspended.
* The automatic watering switch is...odd. I think it should just get removed.
* We should add services for suspending and resuming automatic watering. I think that's the better way to model it in HA.
* Sensors are available again in the new API. I only have a rain sensor, so I can get that implemented, but if someone could give me a dump of what a flow sensor looks like from the API, that would be helpful for modeling that.
* We could add a switch to the controller device that starts and stops all zones
* We could also add a service to the controller for suspending and resuming all zones
* We could add a number entity for the default watering time for each zone. BUT that might be tricky, because the way you set that depends on whether you have "advanced" mode enabled in Hydrawise. Another option is to just ignore that and let users who want to fiddle with this set up automations directly in HA.

Anything else folks are thinking about?

**Comment by smoki3** [comment]
I am tracking this process and already using the config flow. Working fine so far.

I agree completly with your plan. 

For me its important to fix the active suspension sensor. Because it always show "in 2 years". 

Regarding the "automatic watering switch": Does the API differentiate between manual and automatic watering? If yes then it would be fine to have automatic watering as an sensor which shows if a watering is running or not.

Is there a "status per zone" in the API? To see if the zone is suspending or in automatic mode? Would be also nice to have this as an sensor and control it as you mentioned with services.

Regarding the numer entity I also have some doubt. It would be nice to just control the manual watering time without changing the automation. Also a service would be nice to set "manual water for xx minutes".

What do you think are these point possible?

**Comment by thomaskistler** [comment]
Do you still need examples of API responses with flow sensors?

**Comment by dknowles2** [comment]
I think this is now resolved. Feel free to open separate issues for anything else that's missing.
